### ISSUE TITLE
wav2vect+ctc  self-training

### ISSUE THREAD

**Issue by DandelionYoungL** [pytorch/fairseq#3428]
## ❓ Questions and Help
I have some details questions about "Self-training and Pre-training are Complementary for Speech Recognition (Xu et al., 2020)."
In this paper ,after finetune wav2vec+ctc model, it use the self-training methods to finetune on the librivox dataset.  
I have some questions about this.
1.  Is there the self-training methods in fairseq?
2. when i want to get the pseudo labels of librivox dataset, is there some pre-process? e.g. running VAD on the raw wavs? 
3.  If I want to training or infer the wav2vec+ctc model on my own datasets, should I run VAD pre-process on my datasets? And If I use the raw wavs of my dataset( there may some noise or silence), it would work?

**Comment by olafthiele** [comment]
>     1. Is there the self-training methods in fairseq?
You self train first, then you finetune.

>     2. when i want to get the pseudo labels of librivox dataset, is there some pre-process? e.g. running VAD on the raw wavs?
They are not pseudo labels, you need short audio (5-20 secs) with correct transcription.

>     3. If I want to training or infer the wav2vec+ctc model on my own datasets, should I run VAD pre-process on my datasets? And If I use the raw wavs of my dataset( there may some noise or silence), it would work?

Check the Common Voice project, you need input like that. You should have just regular silence. Check Common Voice.

Please close the issue if that answers your questions.
